PublishParasoftTestResults/xsl/cobertura.xsl

```diff
@@ -13,6 +14,118 @@
     </xsl:template>
 
     <xsl:template name="packages">
-        <xsl:element name="packages"></xsl:element>
+        <xsl:element name="packages">
+            <xsl:for-each-group select="/Coverage/Locations/Loc" group-by="substring-before(@uri, tokenize(@uri, '/')[last()])">
+                <xsl:element name="package">
+                    <xsl:attribute name="name">
+                        <xsl:call-template name="addPackageNameAttr"/>
+                    </xsl:attribute>
+                    <xsl:element name="classes">
+                        <xsl:for-each select="current-group()">
+                            <xsl:element name="class">
+                                <xsl:call-template name="addClassFilenameAttr"/>
```

Comment:
Same here for `classFilename`. Let's only generate it once here and save as variable. Then pass in to template which needs this value.